arangod/RestServer/FlushFeature.cpp

```diff
@@ -59,6 +460,123 @@ void FlushFeature::collectOptions(std::shared_ptr<ProgramOptions> options) {
                      arangodb::options::makeFlags(arangodb::options::Flags::Hidden));
 }
 
+/*static*/ bool FlushFeature::registerFlushRecoveryCallback(
+    std::string const& type,
+    FlushRecoveryCallback const& callback
+) {
+  return !callback // skip empty callbacks
+         || getFlushRecoveryCallbacks().emplace(type, callback).second;
```

Comment:
why don't we synchronize access to static variable?
if that's intended, please leave a comment that this particular function is designed to be used in single-threaded environment only

Reply:
please see above